DP-1 cleanup — remove client-authored defaults for server-owned fields

Part of epic #254.

# Child Issue Draft: DP-1 cleanup — remove client-authored defaults for server-owned fields

**Parent Epic:** hyang0129/supreme-claudemander#254
**Child Index:** 1 of 7
**DRI / Team:** TBD
**Rough Effort:** S
**Depends On:** none

---

## Behavioral Question

Does `grep -rE 'starred.*:\s*true' static/index.html` in boot/spawn/broadcast code return zero, and do all #252-tracked e2e failures pass without the client forwarding `starred` as a courier query parameter for paths that originate from a canvas snapshot?

## Inherited Intent (from epic #254)

*Compressed from the validated intent document. Full version: https://github.com/hyang0129/supreme-claudemander/issues/254#issuecomment-4307952397*

**Feared Failure Mode:** Six months from now we discover we need to store client state for a legitimate use case. DP-1 purity has been so aggressively enforced by the #254 migration that the escape hatch — the per-device allowlist in `docs/state-model.md` — is treated as forbidden rather than documented-exception. PRs that add client state get bounced in review. The team (in solo mode: future-self) either shoehorns genuinely device-local state into server state (wasting broadcast bandwidth, creating cross-device flicker for state that should never sync) or secretly stores it in localStorage and hides it from review. The decision that looks wrong in hindsight is treating the per-device allowlist as a last resort rather than a first-class mechanism that #254 should have explicitly reinforced with concrete worked examples.

**Invariants that apply to this slice:**
- The client never invents an authoritative default value for a server-owned field. The `starred: true` fallback at `static/index.html:2028` and every equivalent fallback is removed.

**Relevant Decision Priors:** This slice is an expression of the Dominant Priority — server is the authoritative source of existence and state; client is purely a view/input surface. Any client-side default that chooses a value in the absence of server instruction violates this prior. No other priors apply to this slice.

## Background

The courier pattern added on the #254 branch already forwards `starred` on the `/ws/session/new` query string so the server gets it back verbatim. But the client still has a remaining `starred: true` default at `static/index.html:2028` inside `CanvasClaudeCard.fromDescriptor`, which fires when a server-pushed descriptor omits the field. Every other spawn fallback on the client already defaults to `false` (lines 3223, 3238, 3247, 3257, 3916). The remaining inconsistency is the narrow residue called out in the epic as a DP-1 violation. This slice closes it, closes any other equivalent drift surfaced by the #252 tail, and ships before architectural work begins so hydration does not land on top of a moving foundation.

## Acceptance Scenarios

### Scenario 1 — CanvasClaudeCard.fromDescriptor with missing starred field
**Given** a server-pushed descriptor for a CanvasClaudeCard with `starred` absent (undefined or null)
**When** `CanvasClaudeCard.fromDescriptor(data)` is called in the client
**Then** the reconstructed card has `starred === false` (matching the server's `BaseCard.__init__` default of `False`)
**Falsifiability:** a client-side unit or e2e test that constructs a descriptor without `starred` and asserts the resulting card is not starred; today the test would fail (card is starred:true).

### Scenario 2 — #252 tail e2e failures pass without client-forwarded courier
**Given** the e2e suite as tracked in #252 (15 failures now, trimmed on the active branch — residual failures run in the suite)
**When** the suite runs against a build without the `starred` query-param on `/ws/session/new` originating from snapshot-driven spawns (we keep courier for manually-initiated spawn in this slice)
**Then** every #252 failure is green
**Falsifiability:** running the suite with the courier removed from the snapshot-spawn path today produces the 15 known failures.

### Scenario 3 — Grep sweep returns clean
**Given** the client codebase after this child lands
**When** running `grep -rE 'starred.*:\s*true' claude_rts/static/index.html` ignoring comments
**Then** zero hits in boot, spawn, broadcast-handler, or descriptor-constructor code paths
**Falsifiability:** run the grep today — one hit at line 2028.

## Surface Area

| Component | File / path (with line numbers) | Change required | Scenario(s) it satisfies |
|-----------|--------------------------------|-----------------|--------------------------|
| CanvasClaudeCard.fromDescriptor | `claude_rts/static/index.html:2028` | Change `starred: data.starred != null ? data.starred : true` to `starred: data.starred != null ? data.starred : false` — or delete the fallback entirely if server always emits `starred`. | 1, 3 |
| Grep sweep for any other stale `starred: true` or equivalent client defaults for server-owned fields | `claude_rts/static/index.html` (full file) | Audit + remove any remaining client-authored defaults for `card_uid`, `display_name`, `recovery_script`, `starred` in boot/spawn paths. | 3 |
| #252 tail e2e green | `tests/e2e/test_pr152_starring_rename_recovery.py`, `tests/e2e/test_spawn_registry_e2e.py`, `tests/e2e/test_cross_browser_sync.py` | Run suite, surface any remaining failure, fix at source. No new test authoring required — existing tests are the oracle. | 2 |

## Integration Seams

- `claude_rts/static/index.html` — Children #3, #5, #6 also edit this file but at different line ranges. This child's edits are minimal (single-line fallback flip). Any merge conflict is trivial.
- `tests/e2e/test_pr152_starring_rename_recovery.py` — will be re-touched by Child #3 to update assertions for the inverted spawn path; Child #1 restores baseline correctness first.

## Slice Independence Statement

This slice can ship without any other slice in this epic because it is a pure client-side bug fix against the existing codebase (pre-migration hybrid spawn path). It does not require the new endpoint, hydration, or any server change. Its shipping state is "the existing spawn model, with one fewer DP-1 violation."

## Out of Scope

- Creating the new `GET /api/cards?canvas=X` endpoint — Child #2.
- Changing `/ws/session/new` semantics — Child #3.
- Rendering an "unable to connect" / retry UI — Child #3.
- Any server-side change.

## Open Questions

| ID | Question | Priority | Owner | Deadline |
|----|----------|----------|-------|----------|
| — | None | — | — | — |

---
*Generated by /refine-epic. Review before running /resolve-issue on this child.*

## Refined Spec (generated by /refine-issue)

# Refined Spec: DP-1 cleanup — remove client-authored defaults for server-owned fields (#256)

## Job Statement
When the client constructs a card from a server-pushed descriptor or a canvas snapshot, I want
zero client-authored defaults for any server-owned field, so that the server is the unchallenged
single author of every card's authoritative state and future descriptor-render paths (Child #3) do
not inherit a DP-1 violation from day one.

## Behavioral Intent
After this child lands, `grep -rE 'starred.*:\s*true'` against non-comment production code in
`claude_rts/static/index.html` returns exactly zero hits in card-construction, boot, spawn, or
broadcast-handler paths. The single remaining `starred: true` fallback (in `TerminalCard.fromSerialized`
at line 2028, which is currently dead code but will become live when Child #3 adds the
descriptor-render path) is flipped to `false`, matching the server's `BaseCard.__init__` default.
Separately, the one remaining failing e2e test (`test_state_persists_across_reload`) is resolved,
completing the #252 cluster cleanup so Child #2 begins on a stable green baseline.

## Hidden Assumptions Surfaced
- **Location discrepancy**: The child draft says "CanvasClaudeCard.fromDescriptor at line 2028."
  The actual violation is in `TerminalCard.fromSerialized` (class starts at line 1656;
  `CanvasClaudeCard` starts at 2071 — line 2028 is between them). `fromDescriptor` does not exist
  anywhere in the file; it is a future method planned for Child #3. The implementer must fix
  `TerminalCard.fromSerialized`, not a non-existent method.
- **Currently dead code**: `CARD_TYPE_REGISTRY.deserialize()` has zero callers in the current
  production code, making the `starred: true` at line 2028 unreachable today. The fix is
  preemptive — it closes the violation before Child #3's descriptor-render path makes it live.
- **`spawnFromSerialized` is already clean**: Commit 3526fad already flipped all four
  `starred: false` fallbacks in `spawnFromSerialized` (lines 3223, 3238, 3247, 3257, 3916).
  No changes are needed in that function.
- **Resume button is NOT a violation**: `putCardState(this.sessionId, { starred: true })` at line
  1773 is the resume button's user-initiated state mutation — a `PUT` to the server, not a
  client-authored default. The grep test must exclude this line.
- **One remaining e2e failure**: `test_state_persists_across_reload` in
  `tests/e2e/test_cross_browser_sync.py` currently fails with
  `AssertionError: Session {sid} missing after reload — server snapshot did not replay it`.
  The failure is at `_local_card_id(page_b, sid)` returning `None` after reload — the card's
  session_id is not found in `cards[]`. This is a persistence-across-reload bug separate from
  the `starred: true` default, but it is in the #252 cluster and is within scope of this child
  ("fix at source").
- **No server-side changes needed**: This is a pure client-side fix plus a test fix.

## Acceptance Scenarios

### Entry Point: `TerminalCard.fromSerialized` (static/index.html line 2028)

**Given** a server-pushed descriptor for a `TerminalCard` with `starred` absent (undefined or null)

**When** `TerminalCard.fromSerialized(data)` is called in the client (by
`CARD_TYPE_REGISTRY.deserialize()`, which Child #3 will wire in)

**Then** the reconstructed card has `starred === false` (matching the server's `BaseCard.__init__`
default of `False`)

**Falsifiability:** `grep -rE 'starred.*:\s*true' claude_rts/static/index.html` excluding the
resume-button `putCardState` call returns zero hits. Today it returns one hit at line 2028.

---

### Entry Point: Grep sweep — no other client-authored defaults

**Given** the full `claude_rts/static/index.html` after this child lands

**When** running `grep -rE 'starred.*:\s*true' claude_rts/static/index.html`

**Then** the only match is the resume-button `putCardState` line (line 1773, a user-initiated
server mutation — not a default), or zero matches if the grep excludes comment lines

**Falsifiability:** run `grep -n "starred.*:\s*true" claude_rts/static/index.html | grep -v "//"`
and count non-comment, non-test hits. Before this fix: 1 hit (line 2028). After: 0.

---

### Entry Point: `test_state_persists_across_reload` e2e test

**Given** the stress-test dev-config preset running, with two pages open (A and B)

**When** page A mutates a card's `starred`, `x`, and `y` via `PUT /api/cards/{sid}/state`, then
page B fully reloads (navigates to the page root), and the reload path reads the server-written
canvas snapshot

**Then** after reload, page B finds the same card (same `session_id`) with the mutated position
and starred state — `_local_card_id(page_b, sid)` is non-null and the DOM reflects
`starred=true`, `left=321px`, `top=654px`

**Falsifiability:** running the test currently produces
`AssertionError: Session {sid} missing after reload`. After the fix, it passes.

---

### Entry Point: Full #252-tracked e2e suite green

**Given** the current branch after this child's changes

**When** running `python -m pytest tests/e2e/test_pr152_starring_rename_recovery.py tests/e2e/test_spawn_registry_e2e.py tests/e2e/test_cross_browser_sync.py`

**Then** all tests pass (37+2skip pass, 24 pass, 5+1xfail pass) — zero failures in the #252-tracked files

**Falsifiability:** the current run shows 1 failure (`test_state_persists_across_reload`); the fix
eliminates it.

## Surface Area

Surface area search: examined 5 files + 1 test file. Coverage confidence: **high** — the
`starred: true` violation was already enumerated in the epic; the codebase confirms exactly one
hit remains. The `test_state_persists_across_reload` bug is a separate concern surfaced by
running the suite.

| Component | File / path (with line numbers) | Change required | Scenario(s) it satisfies |
|---|---|---|---|
| `TerminalCard.fromSerialized` | `claude_rts/static/index.html:2028` | Change `starred: data.starred != null ? data.starred : true` to `starred: data.starred != null ? data.starred : false` | Scenario 1, 2 |
| `test_state_persists_across_reload` — session missing after reload | `tests/e2e/test_cross_browser_sync.py:457` | Investigate why `_local_card_id(page_b, sid)` returns `None` after reload. The session `rts-f13dcfb9` is not found in `cards[]` after reload — likely the snapshot written by `apply_state_patch` does not contain the session_id, OR the card's `canvas_map` entry is missing so `_persist_canvas` skips it. Fix at source. | Scenario 3, 4 |

**Note on `test_state_persists_across_reload` root cause investigation:** The test failure trace
shows the assertion fires at line 457 (`b_id_post is not None`) — i.e., after the reload, the
card isn't in `cards[]` at all, not just in the wrong state. The investigator should:
1. Add a `console.log` to verify the canvas snapshot file includes `session_id` after the PUT.
2. Verify the card is in `card_registry._canvas_map` (i.e., `canvas_name` was passed correctly
   when `/ws/session/new` registered it). If `canvas_map` entry is missing, `_persist_canvas`
   silently skips the card and the snapshot never gets `session_id`.
3. Check whether the dev-config `--dev-config stress-test` preset's canvas name
   (`stress-layout`) matches the `currentCanvasName` on the client when `/ws/session/new` is called.

## Out of Scope
- Creating `GET /api/cards?canvas=X` endpoint — Child #2.
- Adding `fromDescriptor` to any card class — Child #3.
- Changing `/ws/session/new` semantics or restricting it to user-initiated spawn — Child #3.
- Rendering a retry / reconnect UI — Child #3.
- Any server-side change to the mutation path or `BaseCard`.
- Adding new test files — existing tests are the oracle.
- Fixing `CanvasClaudeCard.fromSerialized` (it has no `starred` field at all, so no violation there).

## Clarifying Questions
None — proceed directly to implementation. The one pre-implementation step is verifying the
`test_state_persists_across_reload` root cause via the investigation checklist above before
writing the fix.

---
*Spec generated by /refine-issue — ready for review.*

--- a/claude_rts/static/index.html
+++ b/claude_rts/static/index.html
@@ -2025,7 +2025,7 @@ class TerminalCard extends Card {
       symbol: (typeof hubSymbolMap !== 'undefined' && hubSymbolMap[data.hub]) || '?',
       exec: data.exec || null,
       sessionId: data.session_id || null,
-      starred: data.starred != null ? data.starred : true,
+      starred: data.starred != null ? data.starred : false,
       displayName: data.display_name || data.displayName || '',
       recoveryScript: data.recovery_script || data.recoveryScript || '',
       cardUid: data.card_uid || data.cardUid || '',

